src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveScope.cs

```diff
@@ -0,0 +1,674 @@
+    class Class2
+    {
+    }
+}";
+
+            var expected =
+@"namespace N1
+{
+    // Class1 Comment
+    class Class1
+    {
+    }
+}
+
+namespace N1
+{
+    // Class2 Comment
+    class Class2
+    {
+    }
+}";
+
+            return TestNamespaceMove(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public Task MoveType_NamespaceScope_ItemAtTopWithXmlComments()
+        {
+            var code =
+@"namespace N1
+{
+    /// <summary>
+    /// Class1 summary
+    /// </summary>
+    class [||]Class1
+    {
+    }
+
+    /// <summary>
+    /// Class2 summary
+    /// </summary>
+    class Class2
+    {
+    }
+}";
+
+            var expected =
+@"namespace N1
+{
+    /// <summary>
+    /// Class1 summary
+    /// </summary>
+    class Class1
+    {
+    }
+}
+
+namespace N1
+{
+    /// <summary>
+    /// Class2 summary
+    /// </summary>
+    class Class2
+    {
+    }
+}";
+
+            return TestNamespaceMove(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public Task MoveType_NamespaceScope_ItemAtBottom()
+        {
+            var code =
+@"namespace N1
+{
+    class Class1
+    {
+    }
+
+    class [||]Class2
+    {
+    }
+}";
+
+            var expected =
+@"namespace N1
+{
+    class Class1
+    {
+    }
+}
+
+namespace N1
+{
+    class Class2
+    {
+    }
+}";
+
+            return TestNamespaceMove(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public Task MoveType_NamespaceScope_ItemAtBottomWithComments()
+        {
+            var code =
+@"namespace N1
+{
+    // Class1 comment
+    class Class1
+    {
+    }
+
+    // Class2 comment
+    class [||]Class2
+    {
+    }
+}";
+
+            var expected =
+@"namespace N1
+{
+    // Class1 comment
+    class Class1
+    {
+    }
+}
+
+namespace N1
+{
+    // Class2 comment
+    class Class2
+    {
+    }
+}";
+
+            return TestNamespaceMove(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public Task MoveType_NamespaceScope_ItemAtBottomWithXmlComments()
+        {
+            var code =
+@"namespace N1
+{
+    /// <summary>
+    /// Class1 summary
+    /// </summary>
+    class Class1
+    {
+    }
+
+    /// <summary>
+    /// Class2 summary
+    /// </summary>
+    class [||]Class2
+    {
+    }
+}";
+
+            var expected =
+@"namespace N1
+{
+    /// <summary>
+    /// Class1 summary
+    /// </summary>
+    class Class1
+    {
+    }
+}
+
+namespace N1
+{
+    /// <summary>
+    /// Class2 summary
+    /// </summary>
+    class Class2
+    {
+    }
+}";
+
+            return TestNamespaceMove(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public Task MoveType_NamespaceScope_ItemInMiddle()
```

Comment:
Just curious, what should happen if the split namespace has trivia before it?  Would it get copied to all the new namespace declarations?  Or just be left on one. Either way it might be good to have a test case for it.

I can't think of any off the top of my head, but there could be some possible weird edge cases around directives as well that might need to be tested.

Reply:
It won't get copied, since the new declaration is created from scratch with just the same name. That said, a test is an excellent idea. 

I'll try to think more about directive cases, but I think a general rule is for it to be okay that refactoring doesn't work under _abnormal_ conditions. Directives can definitely lead to those, and we don't currently handle weird things like debug vs not. Adding a test that directives remain where they are is worthwhile though.